docker/gcp_pudl_etl.sh

```diff
@@ -100,60 +95,95 @@ function notify_slack() {
 }
 
 function update_nightly_branch() {
-    git config --unset http.https://github.com/.extraheader
-    git config user.email "pudl@catalyst.coop"
-    git config user.name "pudlbot"
-    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git"
-    echo "BOGUS: Updating nightly branch to point at $NIGHTLY_TAG."
-    git fetch origin nightly:nightly
-    git checkout nightly
-    git merge --ff-only "$NIGHTLY_TAG"
-    ETL_SUCCESS=${PIPESTATUS[0]}
-    git push -u origin
+    git config --unset http.https://github.com/.extraheader && \
+    git config user.email "pudl@catalyst.coop" && \
+    git config user.name "pudlbot" && \
+    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git" && \
+    echo "Updating nightly branch to point at $NIGHTLY_TAG." && \
+    git tag -d "$NIGHTLY_TAG" && \
+    git fetch --tags origin "$NIGHTLY_TAG" && \
+    git checkout "$NIGHTLY_TAG" && \
+    git tag && \
+    git fetch origin nightly:nightly && \
+    git checkout nightly && \
+    git merge --ff-only "$NIGHTLY_TAG^0" && \
+    git push -u origin nightly
```

Comment:
Is there anything else in here that we don't need now that it's working?  What does the `^0` do in the emrge?

Reply:
Apparently, [if you merge a tag that *could* be fast-forwarded, `git` will still create a merge commit by default.](https://git-scm.com/docs/git-merge#_merging_tag)

The `^0` "unwraps" the tag, so that git doesn't create a merge commit. But, `--ff-only` should have the same effect, in that it forces a fast-forward merge. I don't think we need to double-dip here, and `--ff-only` is much more legible, so I think we can get rid of `^0`.